src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1345,31 +1347,26 @@ static bool IsOffsetContained(int offset, int start, int length)
 
         void ImportLeave(BasicBlock target)
         {
-            // Empty the stack
-            _stackTop = 0;
+            EmptyTheStack();
 
-#if false
-            for (int i = 0; i < _exceptionRegions.Length; i++)
+            if (_currentBasicBlock.TryIndex.HasValue &&
+                (_exceptionRegions[_currentBasicBlock.TryIndex.Value].ILRegion.Kind == ILExceptionRegionKind.Finally ||
+                 _exceptionRegions[_currentBasicBlock.TryIndex.Value].ILRegion.Kind == ILExceptionRegionKind.Fault))
             {
-                var r = _exceptionRegions[i];
-
-                if (r.ILRegion.Kind == ILExceptionRegionKind.Finally &&
-                    IsOffsetContained(_currentOffset - 1, r.ILRegion.TryOffset, r.ILRegion.TryLength) &&
-                    !IsOffsetContained(target.StartOffset, r.ILRegion.TryOffset, r.ILRegion.TryLength))
-                {
-                    MarkBasicBlock(_basicBlocks[r.ILRegion.HandlerOffset]);
-                }
+                MarkBasicBlock(_basicBlocks[_exceptionRegions[_currentBasicBlock.TryIndex.Value].ILRegion.HandlerOffset]);
```

Comment:
Good point! We talked about it with @mmayr-at and it came up that this should be done somewhere else, but I wanted to go with the original code. Anyway... i will move it to `StartImportingBasicBlock`.